alphagov/frontend
# Reintroduce removed ga4 code

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
- Some GA4 code written by @andysellick in this PR https://github.com/alphagov/frontend/pull/3667 was removed by accident in this PR https://github.com/alphagov/frontend/pull/3724
- This PR reintroduces that code. 
- This PR also removes a duplication of the element to use the partial instead.
- This PR also adds a test to try to prevent this from happening again.

## Why

https://trello.com/c/HNCfXrJG/597-formcomplete-not-firing-when-there-is-no-match-for-services-licence

## How to test
- Run `frontend` locally
- Open `app/controllers/licence_transaction_controller.rb`
- Go to the function `def local_authority_code_from_slug` and make it return `1`
- Open http://127.0.0.1:3005/find-licences/skin-piercing-and-tattooing/islington
- You will see the GA4 attributes in the HTML, along with the GA4 event being sent if you look at the `dataLayer` variable in the JavaScript console.

## Screenshots?

N/A
